src/Files.App/ViewModels/UserControls/Previews/ShellPreviewViewModel.cs

```diff
@@ -112,43 +104,71 @@ private nint WndProc(HWND hwnd, uint msg, nint wParam, nint lParam)
 
 				try
 				{
-					currentHandler = new PreviewHandler(clsid.Value, hwnd.DangerousGetHandle());
-					currentHandler.InitWithFileWithEveryWay(Item.ItemPath);
-					currentHandler.DoPreview();
+					_previewHandler = new PreviewHandler(clsid!.Value, hwnd);
+					_previewHandler.InitWithFileWithEveryWay(Item.ItemPath);
+					_previewHandler.DoPreview();
 				}
 				catch
 				{
 					UnloadPreview();
 				}
 			}
-			else if (msg == (uint)WindowMessage.WM_DESTROY)
+			else if (msg is PInvoke.WM_DESTROY)
 			{
-				if (currentHandler is not null)
+				if (_previewHandler is not null)
 				{
-					currentHandler.Dispose();
-					currentHandler = null;
+					_previewHandler.Dispose();
+					_previewHandler = null;
 				}
 			}
 
-			return DefWindowProc(hwnd, msg, wParam, lParam);
+			return PInvoke.DefWindowProc(hwnd, msg, wParam, lParam);
 		}
 
-		public void LoadPreview(UIElement presenter)
+		public unsafe void LoadPreview(UIElement presenter)
 		{
 			var parent = MainWindow.Instance.WindowHandle;
+			var hInst = PInvoke.GetModuleHandle(default(PWSTR));
+			var szClassName = $"{GetType().Name}-{Guid.NewGuid()}";
+			var szWindowName = $"Preview";
 
-			HINSTANCE hInst = Kernel32.GetModuleHandle();
-
-			wCls = new WindowClass($"{GetType().Name}{Guid.NewGuid()}", hInst, WndProc);
+			fixed (char* pszClassName = szClassName)
+			{
+				_windProc = new(WndProc);
+				var pWindProc = Marshal.GetFunctionPointerForDelegate(_windProc);
+				var pfnWndProc = (delegate* unmanaged[Stdcall]<HWND, uint, WPARAM, LPARAM, LRESULT>)pWindProc;
```

Comment:
I would like to avoid using `GetFunctionPointerForDelegate`. Instead, marking `WndProc` as static and `UnmanagedCallersOnly` and taking a function pointer from it directly.
While we don't need to do it immediately in this PR as it may require some refactor to `WndProc.`

Reply:
Yeah. Since making proc-s static has huge impact like you said, i was hesitant to do this (while I understand using GetFunctionPointerForDelegate has performance impact likewise)

Reply:
Im curious, if you were to, how would you refactor (roughly)?